src/reducer.js

```diff
@@ -37,12 +39,17 @@ function myMerge(oldState, newState) {
         }
 
         const oldValue = result[key];
-        if (isFunction(oldValue.mergeDeep)) {
-            result[key] = oldValue.mergeDeep(value);
-        } else if (isFunction(value.mergeDeep)) {
-            result[key] = fromJS(oldValue).mergeDeep(value);
-        } else if (isObject(value) && !isArray(value)) {
-            result[key] = merge({}, oldValue, value);
+
+        if (!isNull(oldValue) && !isUndefined(oldValue)) {
```

Comment:
@michaelcontento, I guess you right, we able to avoid redundant dependencies here. I'll fix it.